skdownscale/pointwise_models/quantile.py

```diff
@@ -0,0 +1,363 @@
+                'check_pipeline_consistency': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_nan_inf': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_overwrite_params': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_pickle': 'QuantileMappingReressor only suppers 1 feature',
+                'check_fit2d_predict1d': 'QuantileMappingReressor only suppers 1 feature',
+                'check_methods_subset_invariance': 'QuantileMappingReressor only suppers 1 feature',
+                'check_fit2d_1sample': 'QuantileMappingReressor only suppers 1 feature',
+                'check_dict_unchanged': 'QuantileMappingReressor only suppers 1 feature',
+                'check_dont_overwrite_parameters': 'QuantileMappingReressor only suppers 1 feature',
+                'check_fit_idempotent': 'QuantileMappingReressor only suppers 1 feature',
+                'check_n_features_in': 'QuantileMappingReressor only suppers 1 feature',
+                'check_estimators_empty_data_messages': 'skip due to odd sklearn string matching in unit test',
+                'check_regressors_train': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_train(readonly_memmap=True)': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_train(readonly_memmap=True,X_dtype=float32)': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressor_data_not_an_array': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_no_decision_function': 'QuantileMappingReressor only suppers 1 feature',
+                'check_supervised_y_2d': 'QuantileMappingReressor only suppers 1 feature',
+                'check_regressors_int': 'QuantileMappingReressor only suppers 1 feature',
+                'check_methods_sample_order_invariance': 'QuantileMappingReressor only suppers 1 feature',
+            },
+        }
+
+
+class TrendAwareQuantileMappingRegressor(RegressorMixin, BaseEstimator):
+    def __init__(self, qm_estimator=None):
+        self.qm_estimator = qm_estimator
+
+    def fit(self, X, y):
+        self._X_mean_fit = X.mean()
+        self._y_mean_fit = y.mean()
+
+        y_trend = LinearTrendTransformer()
+        y_detrend = y_trend.fit_transform(y)
+
+        X_trend = LinearTrendTransformer()
+        x_detrend = X_trend.fit_transform(X)
+
+        self.qm_estimator.fit(x_detrend, y_detrend)
+
+    def predict(self, X):
+
+        X_trend = LinearTrendTransformer()
+        x_detrend = X_trend.fit_transform(X)
+
+        y_hat = self.qm_estimator.predict(x_detrend).reshape(-1, 1)
+
+        # add the trend back
+        # slope from X (predict)
+        # delta: X (predict) - X (fit) + y
+        trendline = X_trend.lr_model_.coef_[0, 0] * np.arange(len(y_hat)).reshape(-1, 1)
```

Comment:
is `.coef_[0, 0]` the intercept and the slope?

Reply:
or just the intercept?

Reply:
Just the slope. cleaned up and added comments.